src/librand/distributions/gamma.rs

```diff
@@ -103,12 +103,14 @@ impl Gamma {
         assert!(shape > 0.0, "Gamma::new called with shape <= 0");
         assert!(scale > 0.0, "Gamma::new called with scale <= 0");
 
-        let repr = match shape {
-            1.0 => One(Exp::new(1.0 / scale)),
-            0.0...1.0 => Small(GammaSmallShape::new_raw(shape, scale)),
-            _ => Large(GammaLargeShape::new_raw(shape, scale)),
+        let repr = if shape == 1.0 {
+            One(Exp::new(1.0 / scale))
+        } else if shape > 0.0 && shape < 1.0 {
```

Comment:
Should this be `shape >= 0.0`?

(I usually like to write conditions like this with the subject in the middle, e.g., `0.0 <= shape && shape < 1.0`, but YMMV.)